challenges/security/application-security.md

```diff
@@ -0,0 +1,270 @@
+we'll provide relevant links, and you're welcome to ask any questions.
+
+# Rationale
+
+These exercises have several goals:
+
+* It helps us to understand what to expect from you as a developer, what
+  security issues you find important, how you communicate those issues to peers
+  and how you write production code.
+* It helps you get a feel for what it would be like to work at Teleport, as this
+  exercise aims to simulate our day-as-usual and expose you to the type of work
+  and communication you can expect if you join Teleport.
+* We aim to keep the process light. While there are time limits, it is to keep
+  exercises from consuming undue amounts of your time, and not to test
+  performance under pressure.
+
+We believe this technique is not only better, but also is more fun compared to
+whiteboard/quiz interviews so common in the industry.
+
+[Some of the best teams use coding challenges.](https://sockpuppet.org/blog/2015/03/06/the-hiring-post/)
+
+We appreciate your time and are looking forward to the interview.
+
+# Leveling
+There are 6 engineering levels at Teleport. It's possible to earn level 4-5 via
+this interview process.
+
+We are not currently hiring L1-L3 security engineers and level 6 is only
+available via internal promotion. See our
+[security engineering levels](https://github.com/gravitational/careers/blob/main/levels/security.pdf)
+for more details.
+
+We expect different performance at different engineering levels. These
+expectations are explicitly documented below.
+
+# Part 1: Hiring Manager Kick Off
+The hiring manager will discuss your background and approach to security with
+you.
+
+
+# Part 2: Peer Review
+You will be given a GitHub repo containing a PR that includes a
+[RFD](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md)
+and architecture diagram for a service similar to docker.
+
+You will have 90 minutes of asynchronous time to read through the RFD and
+provide written PR feedback in GitHub.
+
+Please be sure to cover security concerns, though you're welcome to provide
+input in any area where you see room for improvement.
+
+## Level 4
+Review the service as written, assuming it will run on a single server.
+
+## Level 5
+The service will run in multiple geographies with unreliable network connections
+between them. Please address the security and business tradeoffs between
+authentication and authorization consistency architectures.
+
+# Part 3: Break
+Grab a bite of food and some water. Take a breather before we continue to the
+next section.
+
+# Part 4: Development
+You will be given a prototype go implementation of the RFD reviewed in
+part 1. You will have 3 hours to develop and submit a single PR implementing
+an authorization and authentication scheme.
+
+This is asynchronous. You'll develop on your machine and you're welcome to
+use whatever tools you're comfortable with to develop the changes. We'll be
+available in the interview slack channel to answer any questions.
```

Comment:
Curious if the Slack channel should be mentioned at the top in the summary with a list of things to expect before the interviewing day?

Reply:
Good suggestion -- I'll make this edit.

Reply:
In the summary, there is the following stanza:

> Throughout the day, the interview team will join you in a slack channel, where we'll provide relevant links, and you're welcome to ask any questions.

Let me know if you'd adjust this wording at all.